File: tests/fixtures/vulnerable/token_no_revocation.ts
// Vulnerable: logout without token invalidation
export function logout(req, res) {
  // No token blacklisting or session destruction
  res.json({ message: "Logged out" });
}

export function resetPassword(req, res) {
  const { newPassword } = req.body;
  updatePassword(req.user.id, newPassword);
  // No token revocation
  res.json({ message: "Password reset" });
}

File: tests/fixtures/vulnerable/token_no_revocation.py
"""Vulnerable: logout/reset handlers without token invalidation."""


def logout(request):
    """Bad: logout without invalidating tokens."""
    return {"message": "Logged out"}


def reset_password(request):
    """Bad: password reset without invalidating old tokens."""
    new_password = request.json["new_password"]
    user = get_user(request.user_id)
    user.password = hash_password(new_password)
    user.save()
    return {"message": "Password reset"}
